cli/src/trade/trade.tsx

```diff
@@ -250,6 +269,180 @@ export const LegMaker = ({
     [buildLeg, setStage, onLegConfirm, addMessage, chosenPairInfo, positionSize]
   )
 
+  const cli = useCli()
```

Comment:
cli option processing and logics could be put in its own component to reduce the code complexity and to focus on core logic here

Reply:
The reason I used cli logic here is to reuse input validation and submit logic

Reply:
Okay. will update